src/configuration/services/solr.md

```diff
@@ -199,6 +199,10 @@ solrsearch:
 
 The Solr 6.x Drupal 8 configuration files are reasonably generic and should work in many other circumstances, but explicitly defining a core, configuration, and endpoint is generally recommended.
 
+### Limitations
+
+Take care to monitor the size of your configuration directories. The `!archive "<directory">` is intended to  gather a collection of related configuration data. The configuration service that backs this is optimised for extremely reliable storage of configuration, like a data dictionary. Typically this involves small collections of key: value stets. As the size of an directory increases, the performance of this data-store will degrade. You should take care to ensure the zipped size of the directory is not too large (anything larger than a couple of Megabytes will perform poorly, and increased past that could contribute to long deployment delays or cause other problems.)
```

Comment:
```suggestion
The recommended maximum size for configuration directories (zipped) is 2MB. These need to be monitored to ensure they don't grow beyond that [how?] If the zipped configuration directories grow beyond this, performance will decline and deploys will become longer. You can reduce the size of the directories by [how?]

The `!archive "<directory">` is a collection of configuration data, like a data dictionary, e.g. small collections of key: value sets. 
```

Reply:
@cmstrickland thanks for this PR. Do we have an example command to monitor these directories? How can the be reduced if they get too big? Thanks.

Reply:
the approximate encoded size of the directory can be seen by running this UNIX pipeline in it `echo $(tar czf - . | base64 | wc -c ) bytes` . Specifically the use of `.` there means 'current directory', this could be replaced with any path.

wrt to reducing them when they get too big, just put less stuff in. Specifically with regards to solr configurations I have seen with problems, it seems like .jar archives for including solr plugins are somtimes being included, these are megabytes in size, and are already compressed.

Reply:
"key: value stets" probably meant to be "sets".  Also, let's use key/value so it's not confused with a grammatical colon.

Reply:
Since plugins are the primary culprit here, let's call that out explicitly and offer some guidance.  "Don't use more than one or two custom plugins", for instance.  I know we offer built-in plugins for many services; it looks like Solr isn't one of them, though.

Perhaps we should consider offering some additional baked-in plugins that people can just enable without having to provide them themselves.

Reply:
I've pushed  updates, but I expect I have included too many words as is my sesquipedalian wont.